### DISCUSSION TITLE
Is it possible to retrieve a single record with Airtable Integration ??

### DISCUSSION THREAD

**Discussion by tresorama** [Budibase/budibase#4330] [Help] [answered]
Hi, is it possible to retrieve a single record by recordId with Airtable Integration?

## What i want to achieve and i can't?

I have two Screen.
First Screen shows all the records of an Airtable Table. I can do it.
Second screen must show a form used to edit a SINGLE record of that Airtable Table.
The UX Flow is that user arrives at first screen, where find a Budibase Table Component with all the records , and by clicking on one row will navigate to the second screen.
First Screen's Table passes "recordID" of the clicked row to second Screen as route param.
When second Screen loads, i trigger the fetch of that single record from Airtable via Data Provider, so i pass the recordId as "Binding" variable to Budibase Query.
But now that "recordId" Binding cannot be sent to Airtable, or at least AIK.

## What i expect to find ?
A text input field where i can inject via HandlebarS syntax the "recordId" Binding.

My airtable table :

<img width="910" alt="Schermata 2022-02-04 alle 17 04 25" src="https://user-images.githubusercontent.com/47954700/152562434-6b84e838-7e9b-4dfc-8bdc-f5181b1da983.png">

PS: the "recordId" field is a formula `RECORD_ID()`.

**Comment by mjashanks** [comment, marked as answer]
Hi there. At the moment, the best way to do this would be to use a transformer to filter your results. This will still return ALL records to budibase, and budibase will filter them.

![image](https://user-images.githubusercontent.com/3524181/152610608-7a3eb2b4-c41a-4870-8eaf-3f2ebbf288a2.png)

If you have a lot of records, this could cause you to have performance issues. If this is an issue, you might be able to overcome this by separating your Airtable into different views, and conditionally select the right view, using a parameter.

**Reply by tresorama** [reply]
## Without params. All records are returned
<img width="698" alt="Schermata 2022-02-05 alle 03 13 11" src="https://user-images.githubusercontent.com/47954700/152625122-9ed03e5c-5fdb-4e05-abf3-56b100d7a7d9.png">
<img width="683" alt="Schermata 2022-02-05 alle 03 14 28" src="https://user-images.githubusercontent.com/47954700/152625128-d14e8c9c-dd6d-4246-9fa3-a098359824f6.png">


## Harcoded param value. it works

<img width="686" alt="Schermata 2022-02-05 alle 03 15 11" src="https://user-images.githubusercontent.com/47954700/152625162-cfb77bc7-e045-46d9-ac42-6daa3e6189c2.png">

## Handlebars  param value. Not working

<img width="662" alt="Schermata 2022-02-05 alle 03 17 32" src="https://user-images.githubusercontent.com/47954700/152625186-08ddd9b6-ab60-4faa-a1eb-0cf981975fe8.png">
<img width="753" alt="Schermata 2022-02-05 alle 03 17 18" src="https://user-images.githubusercontent.com/47954700/152625193-35b0d45a-fda6-42a2-95fd-97814d017c08.png">

### Other attempts
<img width="753" alt="Schermata 2022-02-05 alle 03 22 17" src="https://user-images.githubusercontent.com/47954700/152625433-ce7daf8b-7e0b-47ae-8d72-31052eabb9d4.png">
<img width="753" alt="Schermata 2022-02-05 alle 03 23 41" src="https://user-images.githubusercontent.com/47954700/152625443-de9742cd-271e-4bd2-9659-15cec5f2849b.png">
<img width="753" alt="Schermata 2022-02-05 alle 03 27 59" src="https://user-images.githubusercontent.com/47954700/152625518-dccc2231-f894-4cfa-bdb2-9b4a59aef40f.png">

**Reply by mjashanks** [reply]
Ah apologies - I am misusing bindings....

try

`const prop = params.recordId;`

**Reply by tresorama** [reply]
Thank you @mjashanks , it works !!
I think documentation should clarify better the "Transform Environment" and what is posiible to do.
Maybe also letting console.log in browser console , to grasp if Binding values are accessible or not in the Transform Function.

Anyway , great piece of software, i discovered it yesterday and i like it a lot.

For future reader :

```js
const { recordId } = params;
return data.find(r => r.recordId === recordId)
```

Does `params` contains "Bindings" or Screen route params ?

**Reply by hendrikjap** [reply]
Hi there, do you find the solutions?
